File: backend/FinanceApp.API/Models/Budget.cs
namespace FinanceApp.API.Models;

public class Budget
{
    public int Id { get; set; }
    public int DashboardId { get; set; }
    public int CategoryId { get; set; }
    public decimal Amount { get; set; }
    public BudgetPeriod Period { get; set; } = BudgetPeriod.Monthly;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Dashboard Dashboard { get; set; } = null!;
    public Category Category { get; set; } = null!;
}

public enum BudgetPeriod
{
    Monthly = 0,
    Yearly = 1,
}

File: frontend/src/types/budget.ts
export interface Budget {
  id: number;
  dashboardId: number;
  categoryId: number;
  categoryName: string;
  categoryIcon: string;
  categoryColor: string;
  amount: number;
  period: 'Monthly' | 'Yearly';
  createdAt: string;
}

export interface BudgetProgress {
  budgetId: number;
  categoryId: number;
  categoryName: string;
  categoryIcon: string;
  categoryColor: string;
  budget: number;
  spent: number;
  remaining: number;
  progressPct: number;
  status: 'ok' | 'warning' | 'exceeded';
}

export interface CreateBudget {
  dashboardId: number;
  categoryId: number;
  amount: number;
  period?: 'Monthly' | 'Yearly';
}

export interface UpdateBudget {
  amount?: number;
  period?: 'Monthly' | 'Yearly';
}
